### PR TITLE
Products

### PR DESCRIPTION
**Pull request by jonasdekeukelaere**

### CODE REVIEW

File: Invoice/Item.php [hunk 1]
```diff
@@ -38,6 +38,11 @@ class Item
      */
     protected $discountDescription;
 
+    /**
+     * @var int
```
Comment (carakas) [inline Invoice/Item.php]: int|null

File: Invoice/Item.php [hunk 2]
```diff
@@ -255,6 +260,22 @@ public function setExternalProductId($externalProductId)
         $this->externalProductId = $externalProductId;
     }
 
+    /**
+     * @param int $productId
```
Comment (carakas) [inline Invoice/Item.php]: int|null

File: Invoice/Item.php [hunk 3]
```diff
@@ -255,6 +260,22 @@ public function setExternalProductId($externalProductId)
...
+     */
+    public function setProductId($productId)
+    {
+        $this->productId = $productId;
+    }
+
+    /**
+     * @return int
```
Comment (carakas) [inline Invoice/Item.php]: int|null

File: Product/Product.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+     * @var int|null $id
+     */
+    public function __construct($name, $description, $price, $vat, $id = null)
+    {
+        $this->name = $name;
+        $this->description = $description;
+        $this->price = $price;
+        $this->vat = $vat;
+        if ($id !== null) {
+            $this->id = $id;
```
Comment (carakas) [inline Product/Product.php]: isn't the if statement a bit useless, it is null if the id isn't set anyway?

File: Invoice/Item.php (hunk 1 above)
Comment (jonasdekeukelaere) [reply Invoice/Item.php]: Fixed 2cd3ad2

File: Invoice/Item.php (hunk 2 above)
Comment (jonasdekeukelaere) [reply Invoice/Item.php]: Fixed 2cd3ad2

File: Invoice/Item.php (hunk 3 above)
Comment (jonasdekeukelaere) [reply Invoice/Item.php]: Fixed 2cd3ad2

File: Product/Product.php (hunk 4 above)
Comment (jonasdekeukelaere) [reply Product/Product.php]: Fixed a424da0

**Pull request merged**